plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/utils/AssetPackStateFromDictionaryTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ 	Copyright 2020 Google LLC
+
+ 	Licensed under the Apache License, Version 2.0 (the "License");
+ 	you may not use this file except in compliance with the License.
+ 	You may obtain a copy of the License at
+
+ 		https://www.apache.org/licenses/LICENSE-2.0
+
+ 	Unless required by applicable law or agreed to in writing, software
+ 	distributed under the License is distributed on an "AS IS" BASIS,
+ 	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ 	See the License for the specific language governing permissions and
+ 	limitations under the License.
+*/
+
+package com.google.play.core.godot.assetpacks.utils;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.godotengine.godot.Dictionary;
+
+public class AssetPackStateFromDictionaryTest {
+
+  @Test
+  public void AssetPackStateFromDictionary_valid() {
+    Dictionary testDict =
+        PlayAssetDeliveryUtils.constructAssetPackStateDictionary(
+            42, 0, "awesomePack", 2, 65536, 35);
+    AssetPackStateFromDictionary testSubject = new AssetPackStateFromDictionary(testDict);
+    assertThat(testSubject.bytesDownloaded()).isEqualTo(42);
+    assertThat(testSubject.errorCode()).isEqualTo(0);
+    assertThat(testSubject.name()).isEqualTo("awesomePack");
+    assertThat(testSubject.status()).isEqualTo(2);
+    assertThat(testSubject.totalBytesToDownload()).isEqualTo(65536);
+    assertThat(testSubject.transferProgressPercentage()).isEqualTo(35);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void AssetPackStateFromDictionary_missingKey() {
```

Comment:
Method name should be camelCase (i.e. start with a lower case letter) even though it's referencing a class constructor.

Reply:
Resolved